### ISSUE TITLE
Error response from SQL 'ping' could be more helpful

### ISSUE THREAD

**Issue by candlerb** [zatosource/zato#418]
To reproduce: create an outgoing MySQL connection

* Name: dontcare
* Active: [X]
* Engine: MySQL
* Host: localhost
* Port: 3306
* Database name: testdb
* User: root
* Pool size: 5

Then click "ping". The response in the browser is this:

~~~
Could not ping the outgoing SQL connection, e:[
Traceback (most recent call last):
  File "/opt/zato/2.0.3/zato-web-admin/src/zato/admin/web/views/outgoing/sql.py", line 134, in ping
    response = req.zato.client.invoke('zato.outgoing.sql.ping', {'id':id})
  File "/opt/zato/2.0.3/zato-web-admin/src/zato/admin/middleware.py", line 70, in invoke
    raise Exception(response.inner.text)
Exception: {
  "zato_env": {
    "details": "Traceback (most recent call last):\n
  File \"\/opt\/zato\/2.0.3\/zato-server\/src\/zato\/server\/connection\/http_soap\/channel.py\", line 198, in dispatch\n
    payload, worker_store, self.simple_io_config, post_data)\n
  File \"\/opt\/zato\/2.0.3\/zato-server\/src\/zato\/server\/connection\/http_soap\/channel.py\", line 351, in handle\n
    params_priority=channel_item.params_pri)\n
  File \"\/opt\/zato\/2.0.3\/zato-server\/src\/zato\/server\/service\/__init__.py\", line 423, in update_handle\n
    raise e\nZatoException: Expected elem:[response_time] not found in item:[{u'response_time': u''}]\n",
    "result": "ZATO_ERROR",
    "cid": "K050XE0NGET122WJFKRRX2AE5Y6W"
  }
}]
~~~

However, if you run `tcpdump -i lo -s0 -nnA tcp port 3306` then you see the actual error message returned by the server:

~~~
16:42:10.167559 IP 127.0.0.1.3306 > 127.0.0.1.52170: Flags [P.], seq 96:172, ack 70, win 342, options [nop,nop,TS val 1433538 ecr 1433538], length 76
E.....@.@.................U..Ru\...V.t.....
........H......#28000Access denied for user 'root'@'localhost' (using password: YES)
~~~

It would be helpful if Zato could capture this actual error and return it to the client.

(Aside: it seems strange to me that a SQL ping is invoking "http_soap", but maybe this is part of internal service dispatch path?)

**Comment by dsuch** [comment]
Thanks @candlerb - I can confirm that the ping should be made more explicit as regards the underlying exception, will be working on it now.

As for the the 'http_soap' part, this is only a Python path, part of the call stack,, not name of a service which is 'zato.outgoing.sql.ping'.

**Comment by dsuch** [comment]
Ok, done in 561a4236fbb50c7cbfffc12a3ea7adcd28592e53 and backported to support/2.0 - will be released in 2.0.4 and exceptions will more readable, for instance:

```
Could not ping the outgoing SQL connection, e:[Traceback (most recent call last):
  File "/home/dsuch/projects/zatosource-zato/code/zato-web-admin/src/zato/admin/web/views/outgoing/sql.py", line 134, in ping
    response = req.zato.client.invoke('zato.outgoing.sql.ping', {'id':id})
  File "/home/dsuch/projects/zatosource-zato/code/zato-web-admin/src/zato/admin/middleware.py", line 72, in invoke
    raise Exception('CID: {}\nDetails: {}'.format(zato_env.get('cid'), zato_env.get('details')))
Exception: CID: K05F97DEA2E2AJSG8M5WG2DDG8BQ
Details: Traceback (most recent call last):
  File "/home/dsuch/projects/zatosource-zato/code/zato-server/src/zato/server/connection/http_soap/channel.py", line 198, in dispatch
    payload, worker_store, self.simple_io_config, post_data)
  File "/home/dsuch/projects/zatosource-zato/code/zato-server/src/zato/server/connection/http_soap/channel.py", line 351, in handle
    params_priority=channel_item.params_pri)
  File "/home/dsuch/projects/zatosource-zato/code/zato-server/src/zato/server/service/__init__.py", line 424, in update_handle
    raise e
InterfaceError: (InterfaceError) ('communication error', error(111, 'Connection refused')) None None

]
```
